src/core/evaluator.js

```diff
@@ -2033,9 +2032,8 @@ class PartialEvaluator {
                   localColorSpaceCache,
                 })
                 .then(function (colorSpace) {
-                  if (colorSpace) {
-                    stateManager.state.strokeColorSpace = colorSpace;
-                  }
+                  stateManager.state.strokeColorSpace =
+                    colorSpace || ColorSpace.singletons.gray;
```

Comment:
Similar to the above.
```suggestion
                  if (colorSpace) {
                    stateManager.state.strokeColorSpace = colorSpace;
                  } else if (!stateManager.state.patternStrokeColorSpace) {
                    stateManager.state.strokeColorSpace = ColorSpace.singletons.gray;
                  }
```